### PR TITLE
SB-20892 Delete support for group

### PR DESCRIPTION
**Pull request by sknirmalkar89**

### CODE REVIEW

File: group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    // Get all members belong to the group
+    MemberService memberService = new MemberServiceImpl();
+    List<MemberResponse> membersInDB = memberService.fetchMembersByGroupId(groupId);
+    Response response = groupService.deleteGroup(groupId, membersInDB);
+    // delete cache for the group and all members belong to the group
+    boolean isUseridRedisEnabled =
+        Boolean.parseBoolean(
+            PropertiesCache.getInstance().getConfigValue(JsonKey.ENABLE_USERID_REDIS_CACHE));
+    if (isUseridRedisEnabled) {
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java:76]: Add the code to remove groupId_member cache

File: group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    logger.info("Delete group for the groupId {}", groupId);
+
+    if (StringUtils.isEmpty(userId)) {
+      throw new AuthorizationException.NotAuthorized();
+    }
+    GroupService groupService = new GroupServiceImpl();
+
+    Map<String, Object> dbResGroup = groupService.readGroup(groupId);
+    // Only Group Creator should be able to delete the group
+    if (!userId.equals((String) dbResGroup.get(JsonKey.CREATE_BY))) {
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java:64]: change  this to created_by

File: group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    String groupId = (String) actorMessage.getRequest().get(JsonKey.GROUP_ID);
+    logger.info("Delete group for the groupId {}", groupId);
+
+    if (StringUtils.isEmpty(userId)) {
+      throw new AuthorizationException.NotAuthorized();
+    }
+    GroupService groupService = new GroupServiceImpl();
+
+    Map<String, Object> dbResGroup = groupService.readGroup(groupId);
+    // Only Group Creator should be able to delete the group
```
Comment (reshmi-nair) [inline group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java:63]: Can a group creator delete the group if he is only a member, not an ADMIN? Is there a chance that group creator will be made a normal user?

File: group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java (hunk 2 above)
Comment (sknirmalkar89) [reply group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java]: updated

File: group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java (hunk 3 above)
Comment (sknirmalkar89) [reply group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java:63]: As discussed with Mani.. no one can revoke admin privilege from creator. we don't need to check whether creator will not have admin privilege

Comment (reshmi-nair) [reply group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java:63]: ok

File: group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java (hunk 1 above)
Comment (sknirmalkar89) [reply group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java:76]: updated

**Review by reshmi-nair — APPROVED** [review]

**Pull request merged**